tools/tcpconnect.py

```diff
@@ -228,6 +234,35 @@ def inet_ntoa(addr):
         addr = addr >> 8
     return dq
 
+def inet6_ntoa(addr):
```

Comment:
I think this is overly complicated. An approach with struct.unpack and an array of bytes along with your compacting regex would be a lot less code and still eliminate netaddr dependency. The struct module is also a python stdlib. Something like:

```python
s = ":".join(["%x"%x for x in unpack(">HHHHHHHH", rawaddr)])
s = re.sub(r'(^|:)0:(0:)+', r'::', s, 1)
```